reco_utils/nni/svd_training.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import sys
+
+sys.path.append("../../")
+
+import argparse
+import json
+import logging
+import numpy as np
+import os
+import pandas as pd
+
+import nni
+import surprise
+
+import reco_utils.evaluation.python_evaluation as evaluation
+from reco_utils.recommender.surprise.surprise_utils import compute_rating_predictions, compute_ranking_predictions
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger("surprise_svd")
+
+
+def svd_training(params):
+    """
+    Train Surprise SVD using the given hyper-parameters
+    """
+    logger.debug("Start training...")
+    train_data = pd.read_pickle(path=os.path.join(params['datastore'], params['train_datapath']))
+    validation_data = pd.read_pickle(path=os.path.join(params['datastore'], params['validation_datapath']))
+
+    svd = surprise.SVD(random_state=params['random_state'], n_epochs=params['epochs'], verbose=params['verbose'],
```

Comment:
Can we use **kwargs here? Seems you can simply remove few keys `datastore`, `train_datapath`, `validation_datapath`, `surprise_reader`, `rating_metrics` ... (oh well.. more than I thought)

but would be nice to just pass like: `svb = surprise.SVB(**hyperparams)` if small refactor of this function can do that.